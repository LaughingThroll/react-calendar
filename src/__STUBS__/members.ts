import { VacationTypes } from '../types/model/vacation'

const members = [
  {
    id: 'asdffegegege',
    name: 'Eric Meyer',
    vacations: [
      {
        startDate: '2021-02-12',
        endDate: '2021-02-15',
        type: VacationTypes.PAID,
      },
      {
        startDate: '2021-04-23',
        endDate: '2021-04-29',
        type: VacationTypes.UN_PAID,
      },
      {
        startDate: '2021-04-14',
        endDate: '2021-04-21',
        type: VacationTypes.UN_PAID,
      },
      {
        startDate: '2021-04-01',
        endDate: '2021-04-08',
        type: VacationTypes.PAID,
      },
    ],
  },
  {
    id: 'fllfhllkjj',
    name: 'Chris Coyier',
    vacations: [
      {
        startDate: '2021-01-19',
        endDate: '2021-02-19',
        type: VacationTypes.PAID,
      },
      {
        startDate: '2021-02-20',
        endDate: '2021-02-22',
        type: VacationTypes.UN_PAID,
      },
      {
        startDate: '2021-04-18',
        endDate: '2021-04-28',
        type: VacationTypes.UN_PAID,
      },
      {
        startDate: '2021-03-29',
        endDate: '2021-04-01',
        type: VacationTypes.PAID,
      },
    ],
  },
  {
    id: 'sdaasdwdd',
    name: 'Jack London',
    vacations: [
      {
        startDate: '2021-01-28',
        endDate: '2021-02-01',
        type: VacationTypes.PAID,
      },
      {
        startDate: '2021-01-15',
        endDate: '2021-01-18',
        type: VacationTypes.PAID,
      },
    ],
  },
  {
    id: 'wdwrtettete',
    name: 'Paul Irish',
    vacations: [
      {
        startDate: '2021-02-20',
        endDate: '2021-02-22',
        type: VacationTypes.PAID,
      },
      {
        startDate: '2021-02-25',
        endDate: '2021-03-25',
        type: VacationTypes.PAID,
      },
      {
        startDate: '2021-04-18',
        endDate: '2021-04-28',
        type: VacationTypes.PAID,
      },
    ],
  },
]

export default members
