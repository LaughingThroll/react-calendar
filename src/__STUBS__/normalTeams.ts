import { Team } from '../types/model/team'
import { VacationTypes } from '../types/model/vacation'

export const teams: Team[] = [
  {
    id: 'asdksadowkd',
    members: [
      {
        id: 'asdffegegege',
        name: 'Eric Meyer',
        vacations: [
          {
            startDate: '2021-02-12',
            endDate: '2021-02-15',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-04-23',
            endDate: '2021-04-29',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-14',
            endDate: '2021-04-21',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-01',
            endDate: '2021-04-08',
            type: VacationTypes.PAID,
          },
        ],
      },
      {
        id: 'fllfhllkjj',
        name: 'Chris Coyier',
        vacations: [
          {
            startDate: '2021-01-19',
            endDate: '2021-02-19',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-02-20',
            endDate: '2021-02-22',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-18',
            endDate: '2021-04-28',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-03-29',
            endDate: '2021-04-01',
            type: VacationTypes.PAID,
          },
        ],
      },
      {
        id: 'sdaasdwdd',
        name: 'Jack London',
        vacations: [
          {
            startDate: '2021-01-28',
            endDate: '2021-02-01',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-01-15',
            endDate: '2021-01-18',
            type: VacationTypes.PAID,
          },
        ],
      },
      {
        id: 'wdwrtettete',
        name: 'Paul Irish',
        vacations: [
          {
            startDate: '2021-02-20',
            endDate: '2021-02-22',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-02-25',
            endDate: '2021-03-25',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-04-18',
            endDate: '2021-04-28',
            type: VacationTypes.PAID,
          },
        ],
      },
    ],
    name: 'Frontend Team',
    percentageOfAbsent: [0, 2, 0, 0, 1, 22, 2, 2, 2, 2, 11, 1],
  },
  {
    id: 'owkdowr',
    members: [
      {
        id: 'pppioiopoiiu',
        name: 'Isaac Newton',
        vacations: [
          {
            startDate: '2020-02-20',
            endDate: '2020-02-22',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2020-03-20',
            endDate: '2020-03-22',
            type: VacationTypes.UN_PAID,
          },
        ],
      },
      {
        id: 'qqrtruu',
        name: 'Friedrich Nietzsche',
        vacations: [
          {
            startDate: '2021-03-25',
            endDate: '2021-03-28',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-02-22',
            endDate: '2021-03-22',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-20',
            endDate: '2021-06-09',
            type: VacationTypes.UN_PAID,
          },
        ],
      },
      {
        id: 'rrytyotyoti',
        name: 'Albert Einstein',
        vacations: [
          {
            startDate: '2021-02-15',
            endDate: '2021-03-22',
            type: VacationTypes.UN_PAID,
          },
        ],
      },
    ],
    name: 'Backend Team',
    percentageOfAbsent: [0, 2, 0, 0, 1, 2, 2, 2, 2, 2, 1, 1],
  },
  {
    id: 'skfdgkgjgkdjg',
    members: [
      {
        id: 'ncnxmvn',
        name: 'ROBERT MORRIS',
        vacations: [
          {
            startDate: '2021-03-20',
            endDate: '2021-04-22',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-01-20',
            endDate: '2021-02-22',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-23',
            endDate: '2021-04-29',
            type: VacationTypes.UN_PAID,
          },
        ],
      },
      {
        id: 'sdnnbnvnc',
        name: 'KEVIN MITNINK',
        vacations: [
          {
            startDate: '2021-04-20',
            endDate: '2021-05-12',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-04-14',
            endDate: '2021-04-19',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-01',
            endDate: '2021-04-13',
            type: VacationTypes.PAID,
          },
        ],
      },
    ],
    name: 'QA Team',
    percentageOfAbsent: [16, 14, 0, 0, 1, 2, 0, 2, 2, 0, 0, 28],
  },
  {
    id: 'wdwfetrrrt',
    members: [
      {
        id: 'ncnfjndjgfe',
        name: 'Raymond Kroc',
        vacations: [
          {
            startDate: '2020-03-30',
            endDate: '2020-04-12',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-02-17',
            endDate: '2021-03-20',
            type: VacationTypes.PAID,
          },
          {
            startDate: '2021-04-10',
            endDate: '2021-04-21',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-04-22',
            endDate: '2021-04-25',
            type: VacationTypes.PAID,
          },
        ],
      },
    ],
    name: 'Manager Team',
    percentageOfAbsent: [16, 14, 0, 0, 1, 2, 0, 2, 2, 0, 0, 28],
  },
  {
    id: 'xncmxnmv',
    members: [
      {
        id: 'dsdwdwfgg',
        name: 'Guccio Gucci',
        vacations: [
          {
            startDate: '2021-02-22',
            endDate: '2021-03-15',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-03-30',
            endDate: '2021-04-15',
            type: VacationTypes.PAID,
          },
        ],
      },
      {
        id: 'popoptopp',
        name: 'Gabrielle Bonheur Chanel',
        vacations: [
          {
            startDate: '2021-01-15',
            endDate: '2021-02-28',
            type: VacationTypes.UN_PAID,
          },
          {
            startDate: '2021-03-01',
            endDate: '2021-03-12',
            type: VacationTypes.PAID,
          },
        ],
      },
    ],
    name: 'Design Team',
    percentageOfAbsent: [11, 4, 0, 0, 1, 2, 0, 2, 12, 2, 0, 1],
  },
]
