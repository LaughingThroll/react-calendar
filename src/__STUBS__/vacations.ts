import { VacationTypes } from '../types/model/vacation'
const vacations = [
  {
    startDate: '2021-01-15',
    endDate: '2021-01-18',
    type: VacationTypes.PAID,
  },
  {
    startDate: '2021-01-28',
    endDate: '2021-02-01',
    type: VacationTypes.PAID,
  },
  {
    startDate: '2021-02-12',
    endDate: '2021-02-15',
    type: VacationTypes.PAID,
  },
  {
    startDate: '2021-02-22',
    endDate: '2021-03-22',
    type: VacationTypes.UN_PAID,
  },
  {
    startDate: '2021-03-25',
    endDate: '2021-03-28',
    type: VacationTypes.PAID,
  },
]

export default vacations
