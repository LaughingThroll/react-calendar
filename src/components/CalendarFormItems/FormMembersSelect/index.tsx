import React from 'react'
import { FormItem } from '../../common'
import { Select, Option } from '../../common/Select'
import { findByID } from '../../../utils/forArrays'

import { Team } from '../../../types/model/team'
import { ControlForSelect } from '../../../types/controls'

export interface FormMembersSelectProps {
  teams: Team[]
  teamId: string
  select: ControlForSelect
  classNames?: string[]
}

const FormMembersSelect: React.FC<FormMembersSelectProps> = ({ teams, select, teamId, classNames = [] }) => {
  return (
    <FormItem title="User" classNames={classNames}>
      <Select onChange={select.changeValue} value={select.value} name="members">
        {findByID(teams, teamId)?.members.map(({ name, id }) => (
          <Option key={id} title={name} value={id} />
        ))}
      </Select>
    </FormItem>
  )
}

export default React.memo(FormMembersSelect)
