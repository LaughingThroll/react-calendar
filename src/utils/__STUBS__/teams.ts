import { VacationType } from '../../types/model/vacation'
import { ITeam } from '../../types/model/team'

const teams: ITeam[] = [
  {
    name: 'Frontend Team',
    percentageOfAbsent: [0, 2, 0, 0, 1, 22, 2, 2, 2, 2, 11, 1],
    id: 'asdksadowkd',
    members: [
      {
        name: 'Jack London',
        id: 'sdaasdwdd',
        vacations: [
          {
            startDate: '15.01.2021',
            endDate: '18.01.2021',
            type: VacationType.PAID,
          },
          {
            startDate: '28.01.2021',
            endDate: '01.02.2021',
            type: VacationType.PAID,
          },
        ],
      },
      {
        name: 'Eric Meyer',
        id: 'asdffegegege',
        vacations: [
          {
            startDate: '12.02.2021',
            endDate: '15.02.2021',
            type: VacationType.PAID,
          },
        ],
      },
      {
        name: 'Paul Irish',
        id: 'wdwrtettete',
        vacations: [
          {
            startDate: '20.02.2021',
            endDate: '22.02.2021',
            type: VacationType.PAID,
          },
          {
            startDate: '25.02.2021',
            endDate: '25.03.2021',
            type: VacationType.PAID,
          },
        ],
      },
      {
        name: 'Chris Coyier',
        id: 'fllfhllkjj',
        vacations: [
          {
            startDate: '19.01.2021',
            endDate: '19.02.2021',
            type: VacationType.PAID,
          },
          {
            startDate: '20.02.2021',
            endDate: '22.02.2021',
            type: VacationType.UN_PAID,
          },
        ],
      },
    ],
  },
  {
    name: 'Backend Team',
    percentageOfAbsent: [0, 2, 0, 0, 1, 2, 2, 2, 2, 2, 1, 1],
    id: 'owkdowr',
    members: [
      {
        name: 'Albert Einstein',
        id: 'rrytyotyoti',
        vacations: [
          {
            startDate: '15.02.2021',
            endDate: '22.03.2021',
            type: VacationType.UN_PAID,
          },
        ],
      },
      {
        name: 'Friedrich Nietzsche',
        id: 'qqrtruu',
        vacations: [
          {
            startDate: '22.02.2021',
            endDate: '22.03.2021',
            type: VacationType.UN_PAID,
          },
          {
            startDate: '25.03.2021',
            endDate: '28.03.2021',
            type: VacationType.PAID,
          },
        ],
      },
      {
        name: 'Isaac Newton',
        id: 'pppioiopoiiu',
        vacations: [
          {
            startDate: '20.02.2020',
            endDate: '22.02.2020',
            type: VacationType.UN_PAID,
          },
          {
            startDate: '20.03.2020',
            endDate: '22.03.2020',
            type: VacationType.UN_PAID,
          },
        ],
      },
    ],
  },
]

export default teams
