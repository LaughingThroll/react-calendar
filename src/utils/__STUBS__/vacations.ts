import { VacationType } from '../../types/model/vacation'
const vacations = [
  {
    startDate: '2021-01-15',
    endDate: '2021-01-18',
    type: VacationType.PAID,
  },
  {
    startDate: '2021-01-28',
    endDate: '2021-02-01',
    type: VacationType.PAID,
  },
  {
    startDate: '2021-02-12',
    endDate: '2021-02-15',
    type: VacationType.PAID,
  },
  {
    startDate: '2021-02-22',
    endDate: '2021-03-22',
    type: VacationType.UN_PAID,
  },
  {
    startDate: '2021-03-25',
    endDate: '2021-03-28',
    type: VacationType.PAID,
  },
]

export default vacations
