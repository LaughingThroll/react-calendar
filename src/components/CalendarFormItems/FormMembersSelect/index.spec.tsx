import { shallow } from 'enzyme'
import FormMembersSelect, { FormMembersSelectProps } from './'
import { normalTeams } from '../../../__STUBS__/normalTeams'

const setUp = (props: FormMembersSelectProps) => shallow(<FormMembersSelect {...props} />)

describe('FormMembersSelect render', () => {
  it('correct id default render', () => {
    const props: FormMembersSelectProps = {
      teams: normalTeams,
      teamId: 'asdksadowkd',
      select: { changeValue: jest.fn(), value: '' },
    }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.children().children()).toHaveLength(4)
  })

  it('incorrect id default render', () => {
    const props: FormMembersSelectProps = {
      teams: normalTeams,
      teamId: 'asdkadowkd',
      select: { changeValue: jest.fn(), value: '' },
    }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.children().children()).toHaveLength(0)
  })
})
