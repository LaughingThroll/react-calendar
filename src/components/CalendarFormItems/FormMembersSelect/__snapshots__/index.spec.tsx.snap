// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormMembersSelect render correct id default render 1`] = `
<Memo(FormItem)
  classNames={Array []}
  title="User"
>
  <Memo(Select)
    name="members"
    onChange={[MockFunction]}
    value=""
  >
    <Memo(Option)
      key="asdffegegege"
      title="Eric Meyer"
      value="asdffegegege"
    />
    <Memo(Option)
      key="fllfhllkjj"
      title="Chris Coyier"
      value="fllfhllkjj"
    />
    <Memo(Option)
      key="sdaasdwdd"
      title="Jack London"
      value="sdaasdwdd"
    />
    <Memo(Option)
      key="wdwrtettete"
      title="Paul Irish"
      value="wdwrtettete"
    />
  </Memo(Select)>
</Memo(FormItem)>
`;

exports[`FormMembersSelect render incorrect id default render 1`] = `
<Memo(FormItem)
  classNames={Array []}
  title="User"
>
  <Memo(Select)
    name="members"
    onChange={[MockFunction]}
    value=""
  />
</Memo(FormItem)>
`;
